build.gradle.kts

```diff
@@ -0,0 +1,103 @@
+import org.spongepowered.gradle.task.TaskSortClassMembers
+
+SpongeAPI.project = this
+plugins {
+    `java-library`
+    `maven-publish`
+    // The sponge buildscript will apply various plugins that are
+    // defined in the buildscript classpath (including plugin versions)
+    // that are not listed in Plugins or Versions in SpongeAPI.kt.
+    // They are uniquely defined in buildSrc/build.gradle.kts
+    id(Plugins.sponge)
+    id(Plugins.`event-impl-gen`) version Versions.`event-impl-gen`
+}
+
+apply(plugin = Plugins.spongegradle)
+apply(plugin = Plugins.spongemeta)
```

Comment:
I'm really curious how this doesn't break automatic generated accessors in kotlin dsl. Last time I tried (gradle 4.9 and 4.8) automatic generated accessors wouldn't work if `apply plugin` was used.

Reply:
@Barteks2x because the main gradle plugin is still depended on in the class path, the "sub plugins" are still available to the class path, but not normally applied. Taking advantage of that allows us to apply it and kotlin-dsl generates the accessor.